openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1640,17 +1674,33 @@ static int certificate_cb(SSL* ssl, void* arg) {
         issuers = principalBytes(e, SSL_get_client_CA_list(ssl));
     }
 
-    // Execute the java callback
-    (*e)->CallVoidMethod(e, c->certificate_callback, c->certificate_callback_method,
-             P2J(ssl), types, issuers);
+    // Let's check if we should provide the certificate callback as task that can be run on another Thread.
+    if (c->use_tasks != 0) {
+        // Lets create the CertificateCallbackTask and store it on the SSL object. We then later retrieve it via
+        // SSL.getTask(ssl) and run it.
+        jobject task = (*e)->NewObject(e, certificateCallbackTask_class, certificateCallbackTask_init, P2J(ssl), types, issuers, c->certificate_callback);
+        sslTask = (tcn_ssl_task_t*) OPENSSL_malloc(sizeof(tcn_ssl_task_t));
+        sslTask->task = (*e)->NewGlobalRef(e, task);
```

Comment:
sorry to be nitpicky, but this can fail too.  Normally in Java I wouldn't care so much, but in C the risk is greater and it's much harder to debug.